tfjs-backend-webgpu/src/kernels/BatchMatMul_impl.ts

```diff
@@ -110,35 +110,38 @@ export function batchMatMulImpl({
   // smaller than each of the two input sizes. For example, if input sizes are
   // [12, 2048] and [2048, 1024], the output size is [12, 1024], which is
   // relatively small compared to input sizes.
-  if (!transposeA && !transposeB && ((a.shape[1] <= 16 &&
-      (b.shape[2] <= 512 || b.shape[1] >= 2 * b.shape[2])) ||
-      (b.shape[2] <= 16 &&
-      (a.shape[1] <= 512 || a.shape[2] >= 2 * a.shape[1])))) {
-    program = new MatMulSmallOutputSizeProgram(a3dShape, b3dShape,
-        [batchDim, outerShapeA, outerShapeB], bias, activation,
-        preluActivationWeights);
-  } else if (useVec4) {
-    // TODO: Currently we need to make sure that a.shape[2] and b.shape[2]
-    // are divisible by 4 since we use vec4 to get data. In future, we can
-    // remove this limitation by insert 0 to pack data.
-    program = new MatMulPackedVec4Program(
-        a3dShape, [batchDim, outerShapeA, outerShapeB],
-        env().get('WEBGPU_MATMUL_WORK_PER_THREAD') as number, bias, activation,
-        preluActivationWeights);
-    if ((program as MatMulPackedVec4Program).useWgsl) {
-      const dimAOuter = a3d.shape[1];
-      const dimInner = a3d.shape[2];
-      const dimBOuter = b3d.shape[2];
+  if (!transposeA && !transposeB &&
+      ((a.shape[1] <= 16 &&
+        (b.shape[2] <= 512 || b.shape[1] >= 2 * b.shape[2])) ||
+       (b.shape[2] <= 16 &&
+        (a.shape[1] <= 512 || a.shape[2] >= 2 * a.shape[1])))) {
+    program = new MatMulSmallOutputSizeProgram(
+        a3dShape, b3dShape, [batchDim, outerShapeA, outerShapeB], bias,
+        activation, preluActivationWeights);
+  } else {
+    if (useVec4) {
+      // TODO: Currently we need to make sure that a.shape[2] and b.shape[2]
+      // are divisible by 4 since we use vec4 to get data. In future, we can
+      // remove this limitation by insert 0 to pack data.
+      program = new MatMulPackedVec4Program(
+          a3dShape, [batchDim, outerShapeA, outerShapeB],
+          env().get('WEBGPU_MATMUL_WORK_PER_THREAD') as number, bias,
+          activation, preluActivationWeights);
+    } else {
+      program = new MatMulPackedProgram(
+          a3dShape, [batchDim, outerShapeA, outerShapeB],
+          env().get('WEBGPU_MATMUL_WORK_PER_THREAD') as number, transposeA,
+          transposeB, bias, activation, preluActivationWeights);
+    }
+    if ('useWgsl' in program && program.useWgsl) {
```

Comment:
Move `if (program.useWgsl)` out of the outer if. Otherwise, you need to change it again when you implement MatMulSmallOutputSizeProgram wgsl verson.
```
if (!transposeA && !transposeB && ...) {
} else if (useVec4) {
} else {
}
if (program.useWgsl) {
}

Reply:
done